libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Login/LoginHost/SFSDKLoginHostStorage.m

```diff
@@ -78,35 +80,42 @@ - (id)init {
              * If there are any existing login hosts, remove them as MDM should take
              * highest priority and only the hosts enforced by MDM should be in the list.
              */
-            if([self.loginHostList count] > 0) {
+            if ([self.loginHostList count] > 0) {
                 [self removeAllLoginHosts];
             }
             NSArray *hostLabels = managedPreferences.loginHostLabels;
             [managedPreferences.loginHosts enumerateObjectsUsingBlock:^(NSString *loginHost, NSUInteger idx, BOOL *stop) {
                 NSString *hostLabel = hostLabels.count > idx ? hostLabels[idx] : loginHost;
                 [self.loginHostList addObject:[SFSDKLoginHost hostWithName:hostLabel host:loginHost deletable:NO]];
             }];
-            
-            if(managedPreferences.onlyShowAuthorizedHosts)
+            if (managedPreferences.onlyShowAuthorizedHosts) {
                 return self;
+            }
         }
-        
+
         // Load from info.plist.
-        if([[NSBundle mainBundle] objectForInfoDictionaryKey:@"SFDCOAuthLoginHost"]) {
+        if ([[NSBundle mainBundle] objectForInfoDictionaryKey:@"SFDCOAuthLoginHost"]) {
             NSString *customHost = [[NSBundle mainBundle] objectForInfoDictionaryKey:@"SFDCOAuthLoginHost"];
 
-            // Add the login host from info.plist only if it is not already added.
-            if(![self loginHostForHostAddress:customHost]) {
-                [self.loginHostList addObject:[SFSDKLoginHost hostWithName:customHost host:customHost deletable:NO]];
+            /*
+             * Add the login host from info.plist if it doesn't exist already.
+             * This also handles the case where the custom host configured
+             * was changed between version updates of the application.
+             */
+            if (![self loginHostForHostAddress:customHost]) {
```

Comment:
If the custom host has changed, we should remove the old entries and put in the new custom host value. Since the key is the old host name which we no longer have, we simply remove all of them and add `Production`, `Sandbox`, and the new custom host back in. This does NOT mess with the MDM use case, and applies only to the custom host set through `plist`.